addons/pos_razorpay/static/src/app/services/pos_store.js

```diff
@@ -0,0 +1,52 @@
+import { patch } from "@web/core/utils/patch";
+import { PosStore } from "@point_of_sale/app/services/pos_store";
+
+patch(PosStore.prototype, {
+    async pay() {
+        const currentOrder = this.getOrder();
+        const refundedOrder = currentOrder?.lines[0]?.refunded_orderline_id?.order_id;
+        const razorpayPaymentlines = refundedOrder?.payment_ids.filter(
+            (pi) => pi.payment_method_id.use_payment_terminal === "razorpay"
+        );
+        if (refundedOrder && razorpayPaymentlines?.length === 1) {
+            await super.pay();
+            const paymentIds = refundedOrder.payment_ids || [];
+            // Add all the available payment lines in the refunded order if the current order amount is the same as the refunded order
+            if (Math.abs(currentOrder.getTotalDue()) === refundedOrder.amount_total) {
+                paymentIds.forEach((pi) => {
+                    if (pi.payment_method_id) {
+                        const paymentLine = currentOrder.addPaymentline(pi.payment_method_id);
+                        paymentLine.setAmount(-pi.amount);
+                        paymentLine.updateRefundPaymentLine(pi);
+                    }
+                });
+            } else {
+                // Add available payment lines of refunded order based on conditions.
+                var getTotalDue = currentOrder.getTotalDue();
+                if (getTotalDue < 0 && Math.abs(getTotalDue) > razorpayPaymentlines[0].amount) {
+                    const paymentLine = currentOrder.addPaymentline(
+                        razorpayPaymentlines[0].payment_method_id
+                    );
+                    paymentLine.setAmount(-razorpayPaymentlines[0].amount);
+                    paymentLine.updateRefundPaymentLine(razorpayPaymentlines[0]);
+                }
+                if (currentOrder.getDue() < 0) {
+                    paymentIds.forEach((pi) => {
+                        var current_due = currentOrder.getDue();
+                        if (
+                            current_due < 0 &&
```

Comment:
why do you check this, after the first payment_method that passes the two other conditions, the current_due will be 0 won't it ?

Reply:
Also why do you add the current_due to a payment_method and not the amount that was paid with this payment method ?

Reply:
> why do you check this, after the first payment_method that passes the two other conditions, the current_due will be 0 won't it ?

The current_due will change after adding a payment line, so it needs to be checked each time a payment line is added.